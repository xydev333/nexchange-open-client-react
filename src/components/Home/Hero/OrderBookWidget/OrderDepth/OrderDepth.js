import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { I18n } from 'react-i18next';

import OrderDepthItem from './OrderDepthItem/OrderDepthItem';
import styles from './OrderDepth.scss';


class OrderDepth extends PureComponent {
  state = {
  };

  render() {
    let items = [];
    if(!_.isEmpty(this.props.depth)){
      items = this.props.depth.map((depthItem) => {
        return <OrderDepthItem key={String(depthItem.market)} item={depthItem} side={this.props.side} />;
      });
    }

    return (
      <I18n ns="translations">
        {t => (
          <div className={`${styles.container}`}>
          <h4 className={`${styles.title}`}>{`ORDER DEPTH - ${this.props.side}`}</h4>
          <div className={`${styles.header}`}>
            <span className={``}>{`Market Size (${this.props.selectedCoins.deposit})`}</span>
            <span className={``}>{`Price (1/${this.props.selectedCoins.receive})`}</span>
          </div>
          <div className={``}>
            {items}
          </div>
        </div>
        )}
      </I18n>
    );
  }
}

const mapStateToProps = ({ }) => ({ });
const mapDispatchToProps = dispatch => bindActionCreators({ }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderDepth);
