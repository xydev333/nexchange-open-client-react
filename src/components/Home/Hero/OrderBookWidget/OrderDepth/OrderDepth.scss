.container {
    max-height: 400px;

    .header {
        display: flex;
        height: 40px;
        display: flex;
        align-items: center;
        font-weight: bold;
        font-size: 16px;
        span{
            flex: 1;
        }
    }

    .separator {
        height: 1px;
        margin: 5px 0px;
        background-color: $gray;
    }
}