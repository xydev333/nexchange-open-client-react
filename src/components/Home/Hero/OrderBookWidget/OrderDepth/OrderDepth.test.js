import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import OrderDepth from './OrderDepth.js';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('OrderDepth', () => {
  let wrapShallow;
  const selectedCoins = {receive: 'DOGE', deposit: 'ETH'};
  const sellDepth = [
    { size: 100.000000000, rate: 0.000019500 },
    { size: 100.000000000, rate: 0.000019400 },
    { size: 573.000000000, rate: 0.000017510 }    
    ];
  const buyDepth = [
    { size: 2400.000000000, rate: 0.000015000 },
    { size: 1175.002932550, rate: 0.000017050 },
    { size: 0.000000400, rate: 55666.000000000 }    
    ];

  beforeEach(() => {
    store = mockStore(initialState);
    wrapShallow = shallow(<OrderDepth 
                            store={store}
                            selectedCoins={selectedCoins} 
                            sellDepth={sellDepth}
                            buyDepth={buyDepth} />).dive();
                    });

  it('renders correctly', () => {
    expect(wrapShallow).toMatchSnapshot();
  });

  it('contains 3 sell depth items and 3 buy depth items', () => {
    expect(wrapShallow.find('[data-test="sell-depth-item"]').length).toBe(3);
    expect(wrapShallow.find('[data-test="buy-depth-item"]').length).toBe(3);
  });

  it('has correct spread value', () => {
    let spreadValue;
    const bestBid = sellDepth[sellDepth.length -1];
    const bestAsk = buyDepth[0];

    spreadValue = (bestAsk.rate - bestBid.rate) / ((bestAsk.rate + bestBid.rate)/2) * 100;
    spreadValue = parseFloat(Math.round(spreadValue * 100) / 100).toFixed(2);

    expect(wrapShallow.find('[data-test="spread"]').text()).toEqual(`Spread ${spreadValue}%`);
  });

});
