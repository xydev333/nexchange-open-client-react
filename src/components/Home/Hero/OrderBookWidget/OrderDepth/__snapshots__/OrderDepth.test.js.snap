// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDepth renders correctly 1`] = `undefined`;
